### PR TITLE
use pooled allocator

### PR DESCRIPTION
**Pull request by matthiasdiener**
Reduces memory allocations from 300,000 to 160.
15% speedup.
Addresses #66.

### CODE REVIEW

File: examples/wave-eager.py [hunk 1]
```diff
@@ -54,7 +55,8 @@ def bump(actx, discr, t=0):
 def main():
     cl_ctx = cl.create_some_context()
     queue = cl.CommandQueue(cl_ctx)
-    actx = PyOpenCLArrayContext(queue)
+    actx = PyOpenCLArrayContext(queue,
+        allocator=cl_tools.MemoryPool(cl_tools.ImmediateAllocator(queue)))
```
Comment (inducer) [inline examples/wave-eager.py:59]: It'll be pretty easy to forget that, and as a result get rotten performance. Could you submit a patch to meshmode that produces a warning if a `PyOpenCLArrayContext` is created without an allocator? (where the warning can be silenced by passing `allocator=False`)

Comment (matthiasdiener) [reply examples/wave-eager.py:59]: Done here: https://github.com/inducer/meshmode/pull/34

**Comment by matthiasdiener** [comment]
@inducer Should I do the same for wave-eager-mpi?

**Comment by inducer** [comment]
Yeah

**Comment by inducer** [comment]
Added a bit to #7 to make sure the Euler examples start life with pools enabled as well.

cc @MTCam

**Pull request merged**